### ISSUE TITLE
#️⃣ Topic: Machine Learning

### ISSUE THREAD

**Issue by chuangyu-hscy** [chuangyu-hscy/UoM-DS-2019-2024#2]
# Machine Learning 

This machine learning topic contains content of [COMP20008 Element of Data Preprocessing](https://handbook.unimelb.edu.au/2020/subjects/comp20008) and [COMP30024 Machine Learning](https://handbook.unimelb.edu.au/2021/subjects/comp30027). In addition, this repo also has essential knowledge/algorithms/techniques that I used in [MAST30034 Applied Data Science](https://handbook.unimelb.edu.au/2021/subjects/mast30034). 

Here is the **TODO** list:

### Data Processing Concepts
- [ ] [One Hot Encoding]()
- [ ] [Dummy Encoding]()
- [ ] [Normalisation]()
- [ ] [Standardisation]()
- [ ] [Discretisation]()

### Feature Projection Techniques
- [x] [Principal Component Analysis (PCA)](../tree/main/Machine-Learning/notebooks/PCA.ipynb)
- [ ] [t-distributed stochastic neighbor embedding (t-SNE)]()
- [ ] [Kernel PCA]()
- [ ] [Graph-based kernel PCA]()
- [ ] [Linear Discriminant Analysis (LDA)]()
- [ ] [Generalized Discriminant Analysis (GDA)]()
- [ ] [Autoencoder]()
- [ ] [Missing Values Ratio]()
- [ ] [Low Variance Filter]()
- [ ] [High Correlation Filter]()
- [ ] [Non-negative matrix factorization (NMF)]()
- [ ] [Uniform Manifold Approximation and Projection (UMAP)]()

### Evaluation Metrics
- [x] TP, FP, TN, FN
- [x] Confusion Matrix
- [ ] [Precision, Recall, F1-score]()
- [ ] [Area Under the Curve - Receiver Operating Characteristics (AUC-ROC)]()
- [ ] [Log loss]()
- [ ] [Entropy]()
- [ ] [Mutual Information]()
- [ ] [Information Gain]()
- [ ] [Joint Mutual Information]()
- [ ] [Bootstrap Evaluation]()

### Wrapper Method
- [ ] [Step-wise Forward Feature Selection]()
- [ ] [Backward Feature Elimination]()
- [ ] [Exhaustive Feature Selection]()
- [ ] [Recursive Feature Elimination]()
- [ ] [Boruta]()

### Embeded Methods
- [ ] [Lasso Regularization (L1)]()
- [ ] [Ridge Regularization (L2)]()
- [ ] [Random Forest Importance]()

### Ensemble Learning
- [ ] [Bagging (Bootstrap Aggregating)]()
- [ ] [Boosting]()

### Supervised Learning
- [x] [Regression Analysis](./doc/regression.md)
- [ ] [Linear Regression](./notebooks/supervised/regression/LinearRegression.ipynb)
- [ ] [Multiple Linear Regression](./notebooks/supervised/regression/)  
- [ ] [Logistic Regression](./notebooks/supervised/regression/LogisticRegression.ipynb)
- [ ] [Backward Elimination](./notebooks/supervised/regression/)
- [ ] [Polynomial Regression](./notebooks/supervised/regression/)
- [ ] [Bayesian Linear Regression](./notebooks/supervised/regression/)
- [ ] [Support Vector Regression](./notebooks/supervised/regression/)
- [ ] [Decision Tree Regression](./notebooks/supervised/regression/)
- [ ] [Random Forest Regression](./notebooks/supervised/regression/)
- [ ] [Ridge Regression](./notebooks/supervised/regression/ridge.ipynb)
- [ ] [Lasso Regression](./notebooks/supervised/regression/lasso.ipynb)

### Unsupervised Learning
- [ ] [k-means clustering (KMean)](./notebooks/unsupervised/clustering/KMean.ipynb)
- [ ] [Hierarchical Clustering](./notebooks/unsupervised/clustering/)
    - [ ] Sinlge Linkage
    - [ ] Complete Linkage
    - [ ] Average Linkage
    - [ ] Centroid Linakge
- [ ] [Anomaly detection]()
- [ ] [VAT: Visual Assessment of (Cluster) Tendency](./notebooks/unsupervised/clustering/)
- [ ] [Indenpendent Component Analysis (IDA)]()
- [ ] [Apriori algorithm]()
- [ ] [Singular value decomposition]()
- [ ] [DBSCAN](./notebooks/unsupervised/clustering/)
- [ ] [Mean Shift]()
- [ ] [OPTICS]()
- [ ] [Spectral Clustering]()
- [ ] [Mixture of Gaussians]()
- [ ] [BIRCH]()
- [ ] [Agglomerative Clustering]()
- [ ] [Neural Networks]()
- [ ] [Apriori Algorithm]()
- [ ] [Singular value decomposition]()

### Neural Network
- [ ] [Neuron]()
- [ ] [Layers]()
- [ ] [Epoch]()
- [ ] [Neural Network]()
- [ ] [Convolutional Neural Network]()
- [ ] [Genetic Algorithm]()

### Activation Functions
- [ ] [Linear Activation]()
- [ ] [Heaviside Activation]()
- [ ] [Logistic Activation]()
- [ ] [Sigmoid]()
- [ ] [Rectified Linear Unit (ReLU)]()
- [ ] [tanh]()
- [ ] [Softmax]()
- [ ] [Auto Encoder]()
- [ ] [Genetric Algorithm]()
- [ ] [Ensembler]()
